alpine/FreeElectronLaser.cpp

```diff
@@ -0,0 +1,1851 @@
+    ret.temporal_scale_in_jobfile = get_time_multiplier(j);
+    ret.charge = (config::scalar)j["bunch"]["charge"] * electron_charge_in_unit_charges;
+    ret.mass = (config::scalar)j["bunch"]["mass"] * electron_mass_in_unit_masses;
+    ret.num_particles = (uint64_t)j["bunch"]["number-of-particles"];
+    ret.mean_position  = getVector<config::scalar, 3>(j["bunch"]["position"])                       * lmult / unit_length_in_meters;
+    ret.sigma_position = getVector<config::scalar, 3>(j["bunch"]["sigma-position"])                 * lmult / unit_length_in_meters;
+    ret.position_truncations = getVector<config::scalar, 3>(j["bunch"]["distribution-truncations"]) * lmult / unit_length_in_meters;
+    ret.sigma_momentum = getVector<config::scalar, 3>(j["bunch"]["sigma-momentum"]);
+    ret.output_rhythm = j["output"].contains("rhythm") ? uint32_t(j["output"]["rhythm"]) : 0;
+    ret.output_path = "../data/";
+    if(j["output"].contains("path")){
+        ret.output_path = j["output"]["path"];
+        if(!ret.output_path.ends_with('/')){
+            ret.output_path.push_back('/');
+        }
+    }
+    if(j.contains("experimentation")){
+        nlohmann::json je = j["experimentation"];
+        for(auto it = je.begin(); it!= je.end();it++){
+            ret.experiment_options[it.key()] = double(it.value());
+        }
+    }
+    return ret;
+}
+template<typename scalar>
+using FieldVector = ippl::Vector<scalar, 3>;
+template<typename scalar>
+struct BunchInitialize {
+
+    /* Type of the bunch which is one of the manual, ellipsoid, cylinder, cube, and 3D-crystal. If it is
+     * manual the charge at points of the position vector will be produced.    				*/
+    // std::string     			bunchType_;
+
+    /* Type of the distributions (transverse or longitudinal) in the bunch.				*/
+    std::string distribution_;
+
+    /* Type of the generator for creating the bunch distribution.					*/
+    std::string generator_;
+
+    /* Total number of macroparticles in the bunch.                                                     */
+    unsigned int numberOfParticles_;
+
+    /* Total charge of the bunch in pC.                                                                 */
+    scalar cloudCharge_;
+
+    /* Initial energy of the bunch in MeV.                                                              */
+    scalar initialGamma_;
+
+    /* Initial normalized speed of the bunch.                                                           */
+    scalar initialBeta_;
+
+    /* Initial movement direction of the bunch, which is a unit vector.                                 */
+    FieldVector<scalar> initialDirection_;
+
+    /* Position of the center of the bunch in the unit of length scale.                           	*/
+    // std::vector<FieldVector<scalar> >	position_;
+    FieldVector<scalar> position_;
+
+    /* Number of macroparticles in each direction for 3Dcrystal type.                                   */
```

Comment:
this 3D crystal type thing is inherited from mithra I assume? is it actually supported here? if so I would add some explanation of what it means